libraries/AP_ExternalAHRS/AP_ExternalAHRS_InertialLabs.cpp

```diff
@@ -748,124 +705,124 @@ bool AP_ExternalAHRS_InertialLabs::check_uart()
         (now_usw - last_critical_msg_ms > dt_critical_usw)) {
 
         // Critical USW message
-        if (state2.unit_status & ILABS_UNIT_STATUS_ALIGNMENT_FAIL) {
+        if (state2.unit_status & static_cast<uint16_t>(ILABS_UNIT_STATUS::ALIGNMENT_FAIL)) {
             GCS_SEND_TEXT(MAV_SEVERITY_CRITICAL, "ILAB: Unsuccessful initial alignment");
         }
-        if (state2.unit_status & ILABS_UNIT_STATUS_OPERATION_FAIL) {
+        if (state2.unit_status & static_cast<uint16_t>(ILABS_UNIT_STATUS::OPERATION_FAIL)) {
             GCS_SEND_TEXT(MAV_SEVERITY_CRITICAL, "ILAB: IMU data are incorrect");
         }
 
-        if (state2.unit_status & ILABS_UNIT_STATUS_GYRO_FAIL) {
+        if (state2.unit_status & static_cast<uint16_t>(ILABS_UNIT_STATUS::GYRO_FAIL)) {
             GCS_SEND_TEXT(MAV_SEVERITY_CRITICAL, "ILAB: Gyros failure");
         }
 
-        if (state2.unit_status & ILABS_UNIT_STATUS_ACCEL_FAIL) {
+        if (state2.unit_status & static_cast<uint16_t>(ILABS_UNIT_STATUS::ACCEL_FAIL)) {
             GCS_SEND_TEXT(MAV_SEVERITY_CRITICAL, "ILAB: Accelerometers failure");
         }
 
-        if (state2.unit_status & ILABS_UNIT_STATUS_MAG_FAIL) {
+        if (state2.unit_status & static_cast<uint16_t>(ILABS_UNIT_STATUS::MAG_FAIL)) {
```

Comment:
You can't tell me this looks better!

There are ways this could be done to look quite nice, but this is not one of them!

Reply:
I believe that by using ENUM CLASS, we can detect the bug in #28111 at the coding stage.